crates/jstzd/src/task/octez_node.rs

```diff
@@ -1,25 +1,31 @@
 use super::child_wrapper::{ChildWrapper, SharedChildWrapper};
-use super::Task;
+use super::{file::FileWrapper, Task};
 use anyhow::Result;
 use async_trait::async_trait;
+use octez::r#async::directory::Directory;
 use octez::r#async::endpoint::Endpoint;
 use octez::r#async::node;
 use octez::r#async::node_config::OctezNodeConfig;
-use std::fs::File;
+use std::path::PathBuf;
+use std::sync::Arc;
 
 #[derive(Default, Clone)]
 pub struct OctezNode {
     inner: SharedChildWrapper,
     config: OctezNodeConfig,
+    // holds the TempDir instance so that the directory does not get deleted too soon
+    _data_dir: Arc<Directory>,
+    // holds the NamedTempFile instance so that the file does not get deleted too soon
+    _log_file: Arc<FileWrapper>,
```

Comment:
what about having the `OctezNodeConfig` own the File and Directory? that way we can avoid having these fields and the baker can take both the `OctezNode` and `OctezClient` to initialize

Reply:
I've tried all these yesterday. I don't think `octez::r#async::node:OctezNode` should own the file or the directory since it's simply a wrapper for the executable, and neither should the config because the config carrying file instances sounds way too much responsibility for config, which imo should not be more than a k-v mapping, not to mention that the carried file needs to be in an arc to make the config clonable. `jstzd::task::octez_node::OctezNode`, on the other hand, calls the node executable and thus should manage everything around that, including cleaning up the files or so, and therefore should own the file and the directory. `OctezBaker::with_node_and_client` with the node is better, but it requires access to `octez::r#async::node:OctezNode` instances, which means they need to be somehow exposed when they are not directly accessible, e.g. when there are only `jstzd::task::octez_node::OctezNode` and the node config. Keeping the three setters is sufficient to me without entangling these implicit dependencies for a bit of convenience.

Reply:
Builidng `OctezBakerConfig` from `OctezNodeConfig` and `OctezClient` becomes impossible because `OctezNodeConfig` does not know where the temporary directory is. It used to work because the temporary path was determined by the config builder. Now the temporary path is managed by `jstzd::task::octez_node::OctezNode` and since `OctezNodeConfig` needs to be built before `jstzd::task::octez_node::OctezNode` is spawned, it has no idea what the temporary path is.

There might be some other way of arranging these things and what I'm doing here now is not necessarily tidy enough of course.